### PR TITLE
Resource names pubsub yaml

### PR DESCRIPTION
**Pull request by michaelbausor**

### CODE REVIEW

File: google/pubsub/v1/pubsub_gapic.yaml [hunk 1]
```diff
@@ -83,7 +83,7 @@ interfaces:
     retry_params_name: default
     field_name_patterns:
       name: subscription
-      topic: topic
+      topic: topic_oneof
```
Comment (garrettjonesgoogle) [inline google/pubsub/v1/pubsub_gapic.yaml]: How does this work - what does it apply to?

File: google/pubsub/v1/pubsub_gapic.yaml [hunk 2]
```diff
@@ -450,3 +472,50 @@ interfaces:
...
+    topic: topic
+- message_name: ListTopicsRequest
+  field_entity_map:
+    project: project
+- message_name: ListTopicSubscriptionsRequest
+  field_entity_map:
+    topic: topic
+- message_name: ListTopicSubscriptionsResponse
+  field_entity_map:
+      subscriptions: subscription
```
Comment (garrettjonesgoogle) [inline google/pubsub/v1/pubsub_gapic.yaml]: The indentation is wrong here.

Comment (michaelbausor) [inline google/pubsub/v1/pubsub_gapic.yaml]: Fixed

**Comment by michaelbausor** [comment]
Currently the collection definition is available at the top level, so we could use it right now. My feeling was to wait until we also have the updates to config gen before moving existing collections to the top level, so that regenerating the config doesn't become more difficult that necessary.

**Comment by garrettjonesgoogle** [comment]
ok. LGTM

**Pull request merged**